pkg/apis/duck/v1alpha1/identity_lifecycle.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2020 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"knative.dev/pkg/apis"
+)
+
+func (s *IdentityStatus) MarkWorkloadIdentityConfigured(cs *apis.ConditionSet) {
+	cs.Manage(s).MarkTrue(IdentityConfigured)
+}
+
+func (s *IdentityStatus) MarkWorkloadIdentityNotConfigured(cs *apis.ConditionSet, reason, messageFormat string, messageA ...interface{}) {
+	cs.Manage(s).MarkUnknown(IdentityConfigured, reason, messageFormat, messageA...)
+}
+
+func (s *IdentityStatus) MarkWorkloadIdentityFailed(cs *apis.ConditionSet, reason, messageFormat string, messageA ...interface{}) {
+	cs.Manage(s).MarkFalse(IdentityConfigured, reason, messageFormat, messageA...)
+	// Set ConditionReady to be false.
+	cs.Manage(s).MarkFalse(apis.ConditionReady, "WorkloadIdentityFailed", messageFormat, messageA...)
```

Comment:
Isn't this automatically handled by the ConditionSet? As soon as we call `MarkFalse` on the line above, the Ready condition should automatically be marked as `false`.

Reply:
ConditionType `IdentityConfigured` here is not included in apis.NewLivingConditionSet{} for sources and channel's Type file, so it is not counted for conditionReady. This is because we have two different ways for the credential (secret and WorkloadIdentity), if user choose not to use Workload Identity, but use secret, the instance's condition status will not show `IdentityConfigured` condition.

Reply:
Might be worth adding a comment why we need to set it to False here...

Reply:
I meant add the comment in the code why we do this. Otherwise we will lose the history in the PR comments and is easier for us to see it in the code...